lib/utils.pm

```diff
@@ -1629,4 +1630,10 @@ sub assert_file_content {
     my ($param, $value) = @_;
     assert_script_run("cat $param | grep $value");
 }
+
+# non empty */sys/firmware/efi/* must exist in UEFI mode
+sub is_efi_boot {
+    return !!script_output('test -d /sys/firmware/efi/ && ls -A /sys/firmware/efi/', proceed_on_failure => 1);
```

Comment:
Sure, but the goal is to check whether there is any output produced by `ls -A /sys/firmware/efi/`. Since the first part takes care of the error message in case of missing directory, the other one just prints out the content of the directory.

Reply:
Ok, Then I was just confused by the comment `non empty */sys/firmware/efi/* must exist in UEFI mode`